### DISCUSSION TITLE
Generic props interfaces working well: What happened?

### DISCUSSION THREAD

**Discussion by 0x00A1FA** [vuejs/language-tools#2736] [General]
Hello,

It looks to me like I can use interfaces for props now (example below). What happened? Did 1.6.0 improve something?

```vue
<script setup lang="ts" generic="TModel extends App.Models.Model">

interface Props {
  model: TModel
}

const props = defineProps<Props>()

const model = props.model // Finally working lol
```

I don't get any red squiggly lines when using `withDefaults` or when trying to access the `props` variable.

Thanks!

**Comment by johnsoncodehk** [comment]
There are a lot of changes, but I think specific to your problem it was fixed by https://github.com/vuejs/core/commit/29a9a9bbab059e50fb9e6eb65a159051add44780.
